CMakeLists.txt

```diff
@@ -0,0 +1,60 @@
+cmake_minimum_required(VERSION 3.14 FATAL_ERROR)
+
+project(
+  CPM
+  VERSION 0.37.0
+  LANGUAGES NONE
+)
+
+option(CPM_INSTALL "Install CPM" OFF)
```

Comment:
Is the option necessary? Maybe it is to avoid install on ExternalProject, but if that's the case better use the `if(CMAKE_PROJECT_NAME ...)`

Reply:
I was thinking about people using CPM just to compile their project, forcing install would be unnecessary. But for a framework it could be interesting to allow the install of CPM

Reply:
That is not necessary. Cmake only installs if you pass `cmake --install .`. The only issue that could occur (I neeed to test this), is that calling `cmake --install .` on a downstream project would install the upstreams as well. But I haven't tested that, and if it's not the case, then we're all safe.

Reply:
I think it is the case, CMake will install all even upstreams

Reply:
If you really want to supress this option, maybe we could create a custom target only if we detect CPM is not the main repo and consumer could ask explicitly to install it or make it a dependency to one of the consumer's targets

Reply:
Indeed I confirm this is the case (need to fix my packages). Check the review later on about what is a sufficient way to circumvent this option.